pkg/apis/automationbroker/v1alpha1/types.go

```diff
@@ -161,79 +161,95 @@ const (
 	JobMethodUpdate JobMethod = "update"
 )
 
-// JobStateSpec describes the job state for an async job
-type JobStateSpec struct {
-	State       State     `json:"state"`
-	PodName     string    `json:"podName"`
-	Method      JobMethod `json:"method"`
-	Error       string    `json:"error"`
-	Description string    `json:"description"`
+// Job is a particluar job that is running for a particular bundle type
+type Job struct {
+	Podname          string       `json:"podname"`
+	State            State        `json:"state"`
+	Description      string       `json:"description,omitempty"`
+	Error            string       `json:"error,omitempty"`
+	Method           JobMethod    `json:"method"`
+	LastModifiedTime *metav1.Time `json:"lastModifiedTime,omitempty"`
 }
 
 // +genclient
 // +genclient:noStatus
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// ServiceBinding describes a service binding.
-type ServiceBinding struct {
+// BundleBinding describes a service binding.
+type BundleBinding struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec ServiceBindingSpec `json:"spec"`
+	Spec   BundleBindingSpec   `json:"spec"`
+	Status BundleBindingStatus `json:"status,omitempty"`
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// ServiceBindingList is a list of ServiceBinding resources
-type ServiceBindingList struct {
+// BundleBindingList is a list of ServiceBinding resources
+type BundleBindingList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata"`
 
-	Items []ServiceBinding `json:"items"`
+	Items []BundleBinding `json:"items"`
 }
 
-// ServiceBindingSpec  is a service binding.
-type ServiceBindingSpec struct {
-	ServiceInstanceID string `json:"serviceInstanceID"`
+// BundleBindingSpec  is a service binding.
+type BundleBindingSpec struct {
+	BundleInstance LocalObjectReference `json:"bundleInstance"`
 	// Store the parameters as a json encoded string.
 	Parameters string `json:"parameters"`
-	JobToken   string `json:"jobToken"`
+}
+
+// BundleBindingStatus - status of the bundle
+type BundleBindingStatus struct {
+	State           State          `json:"state"`
+	LastDescription string         `json:"lastDescription,omitempty"`
+	Jobs            map[string]Job `json:"jobs"`
```

Comment:
So if I want the latest Job status, I can find it as the last item in this slice?

Reply:
it's a map, so you would need the token. 

If you didn't know the token, but wanted the last action then you would have to loop through and compare the last modified timestamp.

Reply:
Derp, thank you. That's what I get for skimming this.